app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -192,6 +208,16 @@ export default function createRPCMethodTrackingMiddleware({
         properties.method = method;
       }
 
+      if (process.env.SIWE_V1 && method === MESSAGE_TYPE.PERSONAL_SIGN) {
+        const data = req && req.params && req.params[0];
+        const { isSIWEMessage } = detectSIWE({ data });
+        if (isSIWEMessage) {
+          properties.ui_customizations = [
+            METAMETRIC_KEY_OPTIONS[METAMETRIC_KEY.UI_CUSTOMIZATIONS].SIWE,
+          ];
+        }
+      }
```

Comment:
thinking we could abstract quite a few lines from the custom middleware and next callbacks such that the logic could be easier to read and manage, but this isn't a blocker

Reply:
Yep above 2 blocks of code are exactly similar.